kernel/src/mem/frame_alloc/mod.rs

```diff
@@ -230,14 +231,13 @@ impl CpuLocalFrameCache {
         }
 
         let mut index = 0;
-        let mut base = self.free_list.cursor_front();
-        'outer: while let Some(base_frame) = base.get() {
+        let mut base = self.free_list.iter();
+        'outer: while let Some(base_frame) = base.next() {
             if base_frame.addr().alignment() >= layout.align() {
-                let cursor = base.clone();
                 let mut prev_addr = base_frame.addr();
 
                 let mut c = 0;
-                while let Some(frame) = cursor.get() {
+                for frame in base.by_ref() {
```

Comment:
this actually fixes a bug in the frame allocator that I wasn't aware of, well done!